src/dashboard/src/components/helpers.py

```diff
@@ -300,32 +303,91 @@ def processing_config_path():
     )
 
 
+def _prepare_stream_response(
+    payload,
+    content_type,
+    content_disposition,
+    preview_file=False,
+    already_streaming=True,
+):
+    """Prepare the streaming response to return to the caller."""
+    if preview_file:
+        content_disposition = "inline"
+    if not already_streaming:
+        response = StreamingHttpResponse(payload)
+    else:
+        response = payload
```

Comment:
In the end it's doing the same on both cases. If we always do `response = StreamingHttpResponse(payload)` here and we pass the `stream` variable directly in https://github.com/artefactual/archivematica/pull/1449/files#diff-254823927ae6fbe4584582f13c9b0c18R386, we could remove the `already_streaming` flag. It also makes a little more sense to me to create the `StreamingHttpResponse` always on this function.

Reply:
That makes sense. Once AMClient is released I'll make this change here. Which again, the new tests make much easier to do while proving the efficacy of the fix. Thanks for the suggestions.